extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -360,6 +379,51 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     });
   }
 
+  /**
+   * Handles Fluid-related messages dispatched from SafeFrame.
+   * @param {!Event} event
+   */
+  receiveMessageForFluid(event) {
+    if (event.origin != 'https://tpc.googlesyndication.com') {
+      return;
+    }
+    const data = tryParseJson(getData(event));
+    if (!data || data['s'] != 'creative_geometry_update' ||
+        data['sentinel'] != this.sentinel) {
+      return;
+    }
+    // The first creative_geometry_update message will contain bad
+    // geometric data, as it will have been computed using the initial,
+    // incorrect, iframe style. We use this first message as a
+    // triggering point to restyle the iframe, which will in turn cause
+    // safeframe to send another creative_geometry_update message, this
+    // time containing the correct information.
```

Comment:
Why doesn't the iframe already have proper styles?

Reply:
Because the iframe defaults to a style of 0x0 dimensions.

Reply:
Maybe I'm not understanding the use case. Why isn't the iframe size of the parent element already? Or why isn't the parent element already a set size? Fluid seems to just mean we're gonna change the height later.

Reply:
Technically, they both start with the same size, 0x0. The parent element doesn't have a size, because the idea behind fluid is to alleviate the publisher from having to specify one. Fluid does _sort of_ just mean that we will change the height later, in that the creative will expand to the maximum height given the width that it has to work with. This width is either specified by the pub, or, if not, the maximum available width to it.

Reply:
Alright, then let's split this into two points:

1. The iframe should already have proper styles when it's created. Aren't we already doing `#applyFillContent`?
   - Please remove the iframe set style call.
2. I still don't see why the parent element needs a width? https://jsbin.com/sukubeg/1/edit?html,css,output

Reply:
Okay, I've reworked my solution, and I think you'll be happy(-er) with it. I no longer call computedStyle anywhere. I do however still call setStyle on the iframe, passing a width of '100%' if one is not supplied by the publisher. But I do this inside layoutMeasure. Let me know if you still see any issue with this.